### ISSUE TITLE
[Enhancement]: Filtering within new CoRA prod search results

### ISSUE THREAD

**Issue by lfkennedy** [spawaskar-cora/cora-docs#713]
### Context

Ease of locating specific specimens or subsets of specimens within a Specimen Search list result.

### Description

Ability to filter or sort by individual columns so that similarly coded specimens show up next to each other in the list would enhance search capabilities. 

### Related links

- [Requesting a change](https://docs.coracore.org/en/latest/contributing/requesting-a-change/)
-

### Use Cases

In particular, I could see filtering or sorting by Individual Number, DNA sample number, or Mito Sequence number being helpful as it would group together specimens or make them similarly easier to find in the larger list. Being able to search by provenience 1, for example, would pull up a bunch of specimens that might be associated by provenience, and sorting by Mito Sequence number within that provenience could give an indication of specimens that might be possible to associate to a specific individual.

### Solutions

Sorting "arrows" were available in previous CoRA versions, see circled areas in 2nd photo below.

### Visuals

![image](https://github.com/user-attachments/assets/24d0062a-55f1-48bc-9cb2-4e8c5613ea05)

![image](https://github.com/user-attachments/assets/82099ef7-272d-4288-9844-9abf8c4f1f31)

### Before submitting

- [X] I have read and followed the [change request guidelines](https://docs.coracore.org/en/latest/contributing/requesting-a-change/).
- [X] I have verified that [my idea is a change request and not a bug report](https://docs.coracore.org/en/latest/contributing/requesting-a-change/#its-not-a-bug-its-a-feature).
- [X] I have ensured that, to the best of my knowledge, [my idea will benefit the entire community](https://docs.coracore.org/en/latest/contributing/requesting-a-change/#benefit-for-the-community).
- [X] I have included relevant links to [the documentation](https://docs.coracore.org/), related [issues](https://github.com/spawaskar-cora/cora-docs/issues), and [discussions](https://github.com/spawaskar-cora/cora-docs/discussions) to underline the need for my idea.

**Comment by clegarde** [comment]
I second this request!

**Comment by SachinPawaskarUNO** [comment]
@lfkennedy I agree, this is being planned in the near future.
The data-tables will include sorting and searching (by select fields) in the near future.

Can you maybe list which fields would should have sort and search capabilities in terms of priority order?

**Comment by SachinPawaskarUNO** [comment]
@lfkennedy BTW, Thank you for using the issues template and providing me with all the details along with screenshots.

**Comment by lfkennedy** [comment]
> @lfkennedy I agree, this is being planned in the near future. The data-tables will include sorting and searching (by select fields) in the near future.
> 
> I see that you have provided with an initial list of fields that you would like to see have the sort capability. Can you maybe list which fields would should have sort and search capabilities in terms of priority order?

For me, I am currently updating DNA so the fields I would prioritize would be "DNA Sample Number" and "Mito Sequence number"; "Bone" would be secondary for me
"Individual number" I believe would be helpful in the future but there aren't that many individuals segregated out as of now.

Tagging Carrie in case she has additional priorities for fields that would be nice to have sorting functionality @clegarde

**Comment by clegarde** [comment]
I agree. The following are the important ones, and I think in this general order. 

- Key
- Mito Seq
- DNA Sample
- Individual Number
- Bone

**Comment by clegarde** [comment]
Just to add to this - I've started adding some association info and the fact that we can't sort by "Key" is really frustrating! It's adding a lot more time in data entry. @SachinPawaskarUNO

**Comment by SachinPawaskarUNO** [comment]
@lfkennedy @clegarde For now I have added sortBy capability on Specimen Search screen for Key, Individual Number, External Number and some other specimen fields. Will have to work on sortBy on DNA fields such as Sample Number, Mito Sequence Number.

**Comment by lfkennedy** [comment]
> @lfkennedy @clegarde For now I have added sortBy capability on Specimen Search screen for Key, Individual Number, External Number and some other specimen fields. Will have to work on sortBy on DNA fields such as Sample Number, Mito Sequence Number.

The "key" sort is helpful already, thank you! One question - what does the "1" after the sorted column mean (circled below)?
![image](https://github.com/user-attachments/assets/17a43856-9c80-4473-bcbe-2545a719425b)

**Comment by SachinPawaskarUNO** [comment]
@lfkennedy My plan is to have multi-sort capability for CoRA, which means that you can sort by multiple columns. The 1 in this case would be the first sortBy Column. The subsequent columns would be 2, 3,. ...

Give it a try, First Sort by Side and then by Key within it.

**Comment by lfkennedy** [comment]
@SachinPawaskarUNO cool! It is working for me, that's nice to have. Thanks!

**Comment by CTegtmeyer** [comment]
I believe this issue is resolved
